clang/test/Sema/warn-default-const-init.c

```diff
@@ -0,0 +1,55 @@
+// RUN: %clang_cc1 -fsyntax-only -verify=c,unsafe -Wdefault-const-init %s
+// RUN: %clang_cc1 -fsyntax-only -verify=c,unsafe -Wc++-compat %s
+// RUN: %clang_cc1 -fsyntax-only -verify=unsafe %s
+// RUN: %clang_cc1 -fsyntax-only -verify=c -Wdefault-const-init -Wno-default-const-init-unsafe %s
+// RUN: %clang_cc1 -fsyntax-only -verify=good -Wno-default-const-init-unsafe %s
+// RUN: %clang_cc1 -fsyntax-only -verify=cxx -x c++ %s
+// good-no-diagnostics
+
+struct A { int i; };
+struct S{ const int i; };              // unsafe-note 2 {{member 'i' declared 'const' here}} \
+                                          cxx-note 3 {{default constructor of 'S' is implicitly deleted because field 'i' of const-qualified type 'const int' would not be initialized}}
+struct T { struct S s; };              // cxx-note {{default constructor of 'T' is implicitly deleted because field 's' has a deleted default constructor}}
+struct U { struct S s; const int j; };
```

Comment:
How about 

```cpp
struct Z {
  struct A a;
  const int j;
};
```

To test whether order makes a difference.

Reply:
Sure!